intel_nn_hal/GnaPreparedModel.cpp

```diff
@@ -1460,29 +1512,81 @@ void GnaPreparedModel::asyncExecute(const V1_0_Request& request, MeasureTiming m
         runtimeInfo.unmap_mem();
     }
 
+    return;
+}
+
+std::tuple<ErrorStatus, hidl_vec<V1_2::OutputShape>, Timing>
+ GnaPreparedModel::syncExecute(const V1_3::Request& request, MeasureTiming measure, time_point driverStart) {
+
+#ifdef PERF_COUNTERS
+    runtimeMetrics.infer_calls++;
+#endif
+    time_point driverEnd, deviceStart, deviceEnd;
+    if (measure == MeasureTiming::YES) deviceStart = now();
+
+    //std::vector<RunTimePoolInfo> requestPoolInfos;
+    if (!setRunTimePoolInfosFromHidlMemories(&mRuntimeRequestPoolInfos, request.pools)) {
+        //cb->notify(V1_0::ErrorStatus::GENERAL_FAILURE);
+        return {ErrorStatus::GENERAL_FAILURE, {}, kNoTiming};
+    }
+
+    executeModel(request);
+
+    if (measure == MeasureTiming::YES) {
+        driverEnd = now();
+        Timing timing = {.timeOnDevice = uint64_t(microsecondsDuration(deviceEnd, deviceStart)),
+                         .timeInDriver = uint64_t(microsecondsDuration(deviceEnd, deviceStart))};
+        VLOG(L1, "Driver::asyncExecute timing = %s", toString(timing).c_str());
+    } else {
+        VLOG(L1, "MeasureTiming - No. Returning with no error");
+    }
+
+    return {ErrorStatus::NONE, {}, kNoTiming};
+}
+
+void GnaPreparedModel::asyncExecute(const V1_3::Request& request, MeasureTiming measure, time_point driverStart,
+                          const sp<V1_3::IExecutionCallback>& cb) {
+#ifdef PERF_COUNTERS
+    runtimeMetrics.infer_calls++;
+#endif
+    time_point driverEnd, deviceStart, deviceEnd;
+    if (measure == MeasureTiming::YES) deviceStart = now();
+
+    //std::vector<RunTimePoolInfo> requestPoolInfos;
+    if (!setRunTimePoolInfosFromHidlMemories(&mRuntimeRequestPoolInfos, request.pools)) {
+        cb->notify(V1_0::ErrorStatus::GENERAL_FAILURE);
+        return;
+    }
+
+    executeModel(request);
+
     if (measure == MeasureTiming::YES) {
         driverEnd = now();
         Timing timing = {.timeOnDevice = uint64_t(microsecondsDuration(deviceEnd, deviceStart)),
                          .timeInDriver = uint64_t(microsecondsDuration(deviceEnd, deviceStart))};
         VLOG(L1, "Driver::asyncExecute timing = %s", toString(timing).c_str());
-        cb->notify(V1_0_ErrorStatus::NONE);
+        cb->notify(V1_0::ErrorStatus::NONE);
```

Comment:
@anikulk `cb->notify(V1_0::ErrorStatus::NONE); ` This can be just one line outside the `if else` block.

Reply:
Done